pynestml/utils/channel_processing.py

```diff
@@ -0,0 +1,81 @@
+# -*- coding: utf-8 -*-
+#
+# channel_processing.py
+#
+# This file is part of NEST.
+#
+# Copyright (C) 2004 The NEST Initiative
+#
+# NEST is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# NEST is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+
+from pynestml.utils.mechanism_processing import MechanismProcessing
+
+import sympy
+import re
+
+
+class ChannelProcessing(MechanismProcessing):
+    """Extends MechanismProcessing. Searches for Variables that if 0 lead to the root expression always beeing zero so
+    that the computation can be skipped during the simulation"""
+
+    mechType = "channel"
+
+    def __init__(self, params):
+        super(MechanismProcessing, self).__init__(params)
+
+    @classmethod
+    def collect_information_for_specific_mech_types(cls, neuron, mechs_info):
+        """
+        Searching for parameters in the root-expression that if zero lead to the expression always being zero so that
+        the computation may be skipped.
+        """
+        mechs_info = cls.write_key_zero_parameters_for_root_inlines(mechs_info)
+
+        return mechs_info
+
+    @classmethod
+    def check_if_key_zero_var_for_expression(cls, rhs_expression_str, var_str):
+        if not re.search("1/.*", rhs_expression_str):
+            sympy_expression = sympy.parsing.sympy_parser.parse_expr(rhs_expression_str, evaluate=False)
+            if isinstance(sympy_expression, sympy.core.add.Add) and \
+                    cls.check_if_key_zero_var_for_expression(str(sympy_expression.args[0]), var_str) and \
+                    cls.check_if_key_zero_var_for_expression(str(sympy_expression.args[1]), var_str):
+                return True
+            elif isinstance(sympy_expression, sympy.core.mul.Mul) and \
+                    (cls.check_if_key_zero_var_for_expression(str(sympy_expression.args[0]), var_str) or \
+                    cls.check_if_key_zero_var_for_expression(str(sympy_expression.args[1]), var_str)):
+                return True
+            elif rhs_expression_str == var_str:
+                return True
+            else:
```

Comment:
you can use the "return early" pattern here and eliminate most "else" statements.

Reply:
Don't you think that this would make it very hard to read?